backend/penndata/views.py

```diff
@@ -370,14 +370,14 @@ def get_usage(self, room, date, num_samples, group_by, field):
             usage = self.get_usage_on_date(room, curr, field)  # usage for curr
             # incorporate usage safely considering None (no data) values
             usage_aggs = [
-                (self.safe_add(acc[0], val), acc[1] + (1 if val is not None else 0))
-                for acc, val in zip(usage_aggs, usage)
+                (self.safe_add(sum, val), count + (val is not None))
+                for (sum, count), val in zip(usage_aggs, usage)
             ]
             # update min and max date if any data was logged
             if any(usage):
                 min_date = min(min_date, curr)
                 max_date = max(max_date, curr)
-        ret = [usage_agg[0] / usage_agg[1] if usage_agg[1] else None for usage_agg in usage_aggs]
+        ret = [(sum / count) if count else None for (sum, count) in usage_aggs]
```

Comment:
I believe the crash was because of here when `sum` is `None` but `count` is not `None`. From the `safe_add` change, it looks like `sum` can still be `None` when `count` is not `None`. 

Do we have any thoughts on how to address this?

Reply:
we can probably just modify the conditional to be `if count and sum is not None`

Reply:
if check checks that count is non-zero. Do you have an example of when sum is None but count is not 0? 

The logic is that count can only be incremented when we safe_add a non-None value to sum, and safe_add will never be able to return None if we put in a non-None value (unlike before, hence why the problem before)..

My thinking is count = 0 <=> sum is None, or at least the logic wants it to be that way, so I'd rather not check both and get an error in the case this doesn't happen so we know the logic didn't check out, rather than silently sweeping it under the rug (like if we had checked for both before, I wouldn't have known safe_add is broken)

Reply:
Ah I see. Okay that makes sense. thanks!